src/Importify/Cabal.hs

```diff
@@ -3,28 +3,102 @@
 
 module Importify.Cabal
        ( getLibs
+       , getExtensionMaps
        , readCabal
+       , moduleNameToPath
+       , TargetMap
+       , ExtensionsMap
        ) where
 
-import           Universum
+import           Universum                             hiding (fromString)
 
+import           Data.List                             (nub)
+import qualified Data.Map.Strict                       as Map
+import           Distribution.ModuleName               (ModuleName, fromString,
+                                                        toFilePath)
 import           Distribution.Package                  (Dependency (..), PackageName (..))
-import           Distribution.PackageDescription       (GenericPackageDescription (..),
-                                                        condTreeData, libBuildInfo,
-                                                        targetBuildDepends)
+import           Distribution.PackageDescription       (BuildInfo, Executable,
+                                                        GenericPackageDescription (..),
+                                                        Library, buildInfo,
+                                                        condExecutables, condTreeData,
+                                                        defaultExtensions, exeModules,
+                                                        libBuildInfo, libModules,
+                                                        modulePath, targetBuildDepends)
 import           Distribution.PackageDescription.Parse (readPackageDescription)
 import           Distribution.Verbosity                (normal)
+import           Language.Haskell.Extension            (Extension (..))
+
+type TargetMap = Map String String
+type ExtensionsMap = Map String [String]
```

Comment:
Generally, it's a bad idea to use `String` type because it's very slow. Unfortunately, some libraries still use it. Like `haskell-src-exts`. So usages of `String` type are still unavoidable. But when you're using it a lot, you can convert `String` to `Text` using `toText` function and then work with `Text`. I expect you only need to convert to `Text` only one — from extension name. And then working with map lookups and inserts will be faster as well as file parsing/writing.